ui/app/rif/components/chainAddresses.js

```diff
@@ -0,0 +1,203 @@
+import {DEFAULT_ICON, GET_RESOLVERS, SVG_PLUS} from '../constants';
+import ItemWithActions from './item-with-actions';
+import InputWithSubmit from './InputWithSubmit';
+import rifConfig from '../../../../rif.config';
+import AddNewChainAddressToResolver
+  from '../pages/domainsDetailPage/domainDetailActive/addNewTokenNetworkAddress/addNewChainAddressToResolver';
+import {SLIP_ADDRESSES} from '../constants/slipAddresses';
+
+class ChainAddresses extends Component {
+
+  static propTypes = {
+    domainName: PropTypes.string.isRequired,
+    setChainAddressForResolver: PropTypes.func.isRequired,
+    pageName: PropTypes.string.isRequired,
+    showThis: PropTypes.func.isRequired,
+    redirectParams: PropTypes.any.isRequired,
+    isOwner: PropTypes.bool.isRequired,
+    selectedResolverAddress: PropTypes.string,
+    getChainAddresses: PropTypes.func,
+    newChainAddresses: PropTypes.array,
+    waitForListener: PropTypes.func,
+    showTransactionConfirmPage: PropTypes.func,
+    paginationSize: PropTypes.number,
+    classes: PropTypes.any,
+  }
+
+  constructor (props) {
+    super(props);
+    const resolvers = Object.assign([], GET_RESOLVERS());
+    const slipChainAddresses = Object.assign([], SLIP_ADDRESSES);
+    this.state = {
+      chainAddresses: [],
+      resolvers: resolvers,
+      slipChainAddresses: slipChainAddresses,
+      selectedChainAddress: slipChainAddresses[0].chain,
+      insertedAddress: '',
+      addChainAddress: false,
+    };
+  }
+
+  componentDidMount () {
+    this.loadChainAddresses();
+  }
+
+  componentDidUpdate (prevProps, prevState) {
+    if (prevProps.domainName !== this.props.domainName) {
+      this.loadChainAddresses();
+    } else if (prevProps.newChainAddresses !== this.props.newChainAddresses && this.props.newChainAddresses.length > 0) {
+      this.setState({chainAddresses: this.props.newChainAddresses});
+    }
+  }
+
+  async loadChainAddresses () {
+    if (this.state.resolvers.find(resolver => resolver.address === this.props.selectedResolverAddress && resolver.isMultiChain)) {
+      const chainAddresses = await this.props.getChainAddresses(this.props.domainName);
+      this.setState({chainAddresses: chainAddresses});
+    } else if (rifConfig.mocksEnabled) {
+      const chainAddresses = await this.props.getChainAddresses(this.props.domainName);
+      this.setState({chainAddresses: chainAddresses});
+    }
+  }
+
+  convertChainAddressesToTableData () {
+    const { isOwner, classes } = this.props;
+    return this.state.chainAddresses.map((chainAddress) => {
+      const address = getChainAddressByChainAddress(chainAddress.chain);
+      const icon = address.icon ? address.icon : DEFAULT_ICON;
+      const item = <ItemWithActions contentClasses={classes.content} actionClasses={classes.contentActions} enableEdit={isOwner} enableDelete={isOwner} text={chainAddress.address} leftIcon={icon} onDeleteClick={this.onDeleteClick.bind(this, chainAddress.chain)}>
```

Comment:
Be careful with this, you should put the component declaration in jsx sintax, that means you need to put a open parenthesis and close parenthesis at the end instead.

Reply:
I think thats depends on how you use it later. 

If you use `data={item} `is ok.